http-server/src/main/clojure/xtdb/server.clj

```diff
@@ -104,6 +106,33 @@
             (let [writer (transit/writer out :json opts)]
               (transit/write writer res))))))))
 
+(defn- ->jsonl-resultset-encoder [_opts]
+  (reify
+    mf/EncodeToBytes
+    ;; we're required to be a sub-type of ETB but don't need to implement its fn.
+
+    mf/EncodeToOutputStream
+    (encode-to-output-stream [_ res _]
+      (fn [^OutputStream out]
+        (if-not (ex-data res)
+          (with-open [^IResultSet res res
+                      writer (io/writer out)]
+            (try
+              (doseq [el (iterator-seq res)]
+                (json/write el writer)
+                (.write writer "\n"))
+              ;; TODO how to to mark it as an error for the client
```

Comment:
the response status will likely do this.

in addition, I'd be tempted to return `{:message (.getMessage e), :data (ex-data e)}` here - think you might not even need a special case for `xtdb.RuntimeException` if `ex-data` handles non-IExceptionInfos gracefully

Reply:
regarding repsonse status, see my comment below.

Reply:
```clj
(try 
  (/ 1 0)
  (catch Throwable t
    {:message (.getMessage t) :data (ex-data t)}))
;; => {:message "Divide by zero", :data nil}

(try 
  (/ 1 0)
  (catch Throwable t
    (ex-data (throwable->ex-info t))))
;; => {:xtdb.error/error-type :unknown-runtime-error,
;;     :class "java.lang.ArithmeticException",
;;     :stringified "java.lang.ArithmeticException: Divide by zero"}
```
I'd prefer the latter.

Reply:
We are going for one error type for all of JSON.